sbi/inference/posteriors/base_posterior.py

```diff
@@ -302,24 +400,39 @@ def _prepare_for_sample(
                 Importance-Resampling using `init_strategy_num_candidates` to find init
                 locations.
 
-        Returns: Single (potentially default) $x$ with batch dimension; an integer
-            number of samples; a (potentially default) mcmc method; and (potentially
+        Returns: A (potentially default) mcmc method and (potentially
             default) mcmc parameters.
         """
-
-        x = atleast_2d_float32_tensor(self._x_else_default_x(x))
-        if not self._allow_iid_x:
-            self._ensure_single_x(x)
-        self._ensure_x_consistent_with_default_x(x)
-        num_samples = torch.Size(sample_shape).numel()
-
         mcmc_method = mcmc_method if mcmc_method is not None else self.mcmc_method
         mcmc_parameters = (
             mcmc_parameters if mcmc_parameters is not None else self.mcmc_parameters
         )
+        return mcmc_method, mcmc_parameters
 
-        # Move x to current device.
-        return x.to(self._device), num_samples, mcmc_method, mcmc_parameters
+    def _potentially_replace_rejection_parameters(
```

Comment:
Hm. It does not `set` them. The logic is:
```python
a = a if a is not None else self._a
```
So, we really **replace**, and not set.

Reply:
I see, but I still find "potentially" too long / confusing. how about "maybe" instead of "potentially" or just `_get_rejection_parameters` or `_replace_default_rejection_parameters_if_passed`

Reply:
`maybe` sounds good to me!